Library/Homebrew/brew.sh

```diff
@@ -191,6 +191,16 @@ case "$HOMEBREW_COMMAND" in
   --config)    HOMEBREW_COMMAND="config" ;;
 esac
 
+if [[ -z "$HOMEBREW_DEVELOPER" ]]
+then
+  export HOMEBREW_GIT_CONFIG_FILE="$HOMEBREW_REPOSITORY/.git/config"
+  HOMEBREW_GIT_CONFIG_DEVELOPERMODE="$(git config --file="$HOMEBREW_GIT_CONFIG_FILE" --get homebrew.developermode)"
```

Comment:
Good question. Basically because we want `brew update` and other things that change behaviour (e.g. printing backtraces) to be different for Homebrew developers.